app/src/androidTest/java/com/sample/neuroid/us/MockServerTest.kt

```diff
@@ -0,0 +1,148 @@
+                    reader.endArray()
+                    jsonTokens.push(reader.peek())
+                }
+                JsonToken.BEGIN_OBJECT -> {
+                    reader.beginObject()
+                    jsonTokens.push(reader.peek())
+                }
+                JsonToken.END_OBJECT -> {
+                    reader.endObject()
+                    jsonTokens.push(reader.peek())
+                }
+                JsonToken.NAME ->  {
+                    val t = reader.nextName()
+                    if (t == eventType) {
+                        booleanIsFound = true
+                    } else {
+                        jsonTokens.push(reader.peek())
+                    }
+                }
+                JsonToken.STRING -> {
+                    val t = reader.nextString()
+                    if (t == eventType) {
+                        booleanIsFound = true
+                    } else {
+                        jsonTokens.push(reader.peek())
+                    }
+                }
+                JsonToken.NUMBER -> {
+                    val t = reader.nextString()
+                    if (t == eventType) {
+                        booleanIsFound = true
+                    } else {
+                        jsonTokens.push(reader.peek())
+                    }
+                }
+                JsonToken.BOOLEAN -> {
+                    reader.nextBoolean()
+                    jsonTokens.push(reader.peek())
+                }
+                JsonToken.NULL -> {
+                    reader.nextNull()
+                    jsonTokens.push(reader.peek())
+                }
+                JsonToken.END_DOCUMENT -> {
+                    // end of doc, we should exit here
+                }
+            }
+        }
+    }
+
+    /**
+     * switched from a DOM model to a stream to reduce memory usage when parsing the
+     * JSON response. so far this is making the tests more reliable locally.
+     */
+    fun assertRequestBodyContains(eventType:String) {
+        var request = server.requestCount
+        booleanIsFound = false
+        if (request > 0) {
+            for (i in 0 until request) {
+                var req = server.takeRequest()
+                val body = req.body.readUtf8()
+                val gson = GsonBuilder()
+                    .registerTypeAdapter(LocationListener::class.java, LocationListenerCreator())
+                    .registerTypeAdapter(CoroutineScope::class.java, CoroutineScopeAdapter())
+                    .create()
+                val reader = gson.newJsonReader(StringReader(body))
+                reader?.let {
+                    processJSONObject(it, eventType)
+                }
+            }
+
+            if (booleanIsFound) {
+                assert(true) {
+                    "Found the event man!"
+                }
+            } else {
+                assert(false) {
+                    "Failed to find event man!"
```

Comment:
Should we put the event type that was not found?

Reply:
can do